src/backend/distributed/commands/utility_hook.c

```diff
@@ -276,6 +275,21 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 				UndistributeDisconnectedCitusLocalTables();
 			}
 			ResetConstraintDropped();
+
+			if (AddAllLocalTablesToMetadata && !IsBinaryUpgrade &&
+				IsA(parsetree, CreateStmt) && context == PROCESS_UTILITY_TOPLEVEL &&
+				IsCoordinator() && CoordinatorAddedAsWorkerNode())
```

Comment:
Would be good to define a function that captures all of these conditions and the ones below.